disperser/dataapi/v2/server_v2.go

```diff
@@ -594,6 +617,127 @@ func (s *ServerV2) FetchBlobInclusionInfoHandler(c *gin.Context) {
+//	@Success	200				{object}	OperatorsSigningInfoResponse
+//	@Failure	400				{object}	ErrorResponse	"error: Bad request"
+//	@Failure	404				{object}	ErrorResponse	"error: Not found"
+//	@Failure	500				{object}	ErrorResponse	"error: Server error"
+//	@Router		/operators/signing-info [get]
+func (s *ServerV2) FetchOperatorSigningInfo(c *gin.Context) {
+	timer := prometheus.NewTimer(prometheus.ObserverFunc(func(f float64) {
+		s.metrics.ObserveLatency("FetchOperatorSigningInfo", f*1000) // make milliseconds
+	}))
+	defer timer.ObserveDuration()
+
+	var err error
+
+	now := time.Now()
+	oldestTime := now.Add(-maxBlobAge)
+
+	endTime := now
+	if c.Query("end") != "" {
+		endTime, err = time.Parse("2006-01-02T15:04:05Z", c.Query("end"))
+		if err != nil {
+			s.metrics.IncrementInvalidArgRequestNum("FetchOperatorSigningInfo")
+			invalidParamsErrorResponse(c, fmt.Errorf("failed to parse end param: %w", err))
+			return
+		}
+		if endTime.Before(oldestTime) {
+			s.metrics.IncrementInvalidArgRequestNum("FetchOperatorSigningInfo")
+			invalidParamsErrorResponse(
+				c, fmt.Errorf("end time cannot be more than 14 days in the past, found: %s", c.Query("end")),
+			)
+			return
+		}
+	}
+
+	interval := 3600
+	if c.Query("interval") != "" {
+		interval, err = strconv.Atoi(c.Query("interval"))
+		if err != nil {
+			s.metrics.IncrementInvalidArgRequestNum("FetchOperatorSigningInfo")
+			invalidParamsErrorResponse(c, fmt.Errorf("failed to parse interval param: %w", err))
+			return
+		}
+		if interval <= 0 {
+			s.metrics.IncrementInvalidArgRequestNum("FetchOperatorSigningInfo")
+			invalidParamsErrorResponse(c, fmt.Errorf("interval must be greater than 0, found: %d", interval))
+			return
+		}
+	}
+
+	quorumStr := "0,1"
+	if c.Query("quorums") != "" {
+		quorumStr = c.Query("quorums")
+	}
+	quorums := strings.Split(quorumStr, ",")
+	quorumsSeen := make(map[uint8]struct{}, 0)
+	for _, idStr := range quorums {
+		id, err := strconv.Atoi(idStr)
+		if err != nil {
+			s.metrics.IncrementInvalidArgRequestNum("FetchOperatorSigningInfo")
+			invalidParamsErrorResponse(c, fmt.Errorf("failed to parse the provided quorum: %s", quorumStr))
+			return
+		}
+		if id < 0 || id > maxQuorumIDAllowed {
+			s.metrics.IncrementInvalidArgRequestNum("FetchOperatorSigningInfo")
+			invalidParamsErrorResponse(
+				c, fmt.Errorf("the quorumID must be in range [0, %d], found: %d", maxQuorumIDAllowed, id),
+			)
+			return
+		}
+		quorumsSeen[uint8(id)] = struct{}{}
+	}
+	quorumIds := make([]uint8, 0, len(quorumsSeen))
+	for q := range quorumsSeen {
+		quorumIds = append(quorumIds, q)
+	}
+
+	nonsignerOnly := "false"
+	if c.Query("nonsigner_only") != "" {
+		nonsignerOnly = c.Query("nonsigner_only")
+		if nonsignerOnly != "true" && nonsignerOnly != "false" {
+			invalidParamsErrorResponse(c, errors.New("the nonsigner_only param must be \"true\" or \"false\""))
+			return
+		}
+	}
+
+	startTime := endTime.Add(-time.Duration(interval) * time.Second)
```

Comment:
Should we add to the documentation that the API can only compute the signing info within the 14 day period (maxBlobAge)?

Reply:
It's a system wide constraint (no data is guaranteed after 14 days), not anything special

Reply:
Just want to make sure it's clear to the consumer of the Data API that they shouldn't expect to be able to query beyond the 14 days. Some may be interested in historical signing rate beyond the 14 days.